store/sqlstore/channel_store.go

```diff
@@ -299,21 +320,112 @@ func (s SqlChannelStore) CreateIndexesIfNotExists() {
 	s.CreateIndexIfNotExists("idx_channelmembers_user_id", "ChannelMembers", "UserId")
 
 	s.CreateFullTextIndexIfNotExists("idx_channel_search_txt", "Channels", "Name, DisplayName, Purpose")
+
+	s.CreateIndexIfNotExists("idx_publicchannels_team_id", "PublicChannels", "TeamId")
+	s.CreateIndexIfNotExists("idx_publicchannels_name", "PublicChannels", "Name")
+	s.CreateIndexIfNotExists("idx_publicchannels_delete_at", "PublicChannels", "DeleteAt")
+	if s.DriverName() == model.DATABASE_DRIVER_POSTGRES {
+		s.CreateIndexIfNotExists("idx_publicchannels_name_lower", "PublicChannels", "lower(Name)")
+		s.CreateIndexIfNotExists("idx_publicchannels_displayname_lower", "PublicChannels", "lower(DisplayName)")
+	}
```

Comment:
Remind me why this is only necessary for Postgres?

Reply:
This particular index would be a performance optimization for use in `buildLIKEClause`, where we also special-case Postgres behaviour to match on `lower(%s)` instead of just `%s`. 

However, I'm uncertain /why/ we do lower-case matching for Postgres and not for MySQL, since -- unless I'm misunderstanding -- that amounts to a functional difference in sorting behaviour depending on your database of choice.

At this point, we're just preserving same, but it might be worth investigating further.